cmd/xl-v1-list-objects-heal.go

```diff
@@ -325,15 +325,14 @@ func (xl xlObjects) listMultipartUploadsHeal(bucket, prefix, keyMarker,
 		}
 		// Collect uploads until maxUploads limit is reached.
 		for walkResult := range walkerCh {
-			// Ignore errors like errDiskNotFound
-			// and errFileNotFound.
-			if isErrIgnored(walkResult.err,
-				xlTreeWalkIgnoredErrs...) {
-				continue
-			}
-			// For any error during tree walk we should
-			// return right away.
+
+			// For any error during tree walk, except for file not found,
```

Comment:
@vadmeste `listDirFactory` shouldn't return an ignored error. `errFileNotFound` is an ignored error.  See PR https://github.com/minio/minio/pull/3935. With that being the case, we won't be able to check if `err == errFileNotFound`. The fix is to add the following snippet after the for loop over the `walkResultCh`. This checks if the for-loop was exited on iterating through all entries or on `maxUploads` becoming `0`.
```go
if maxUploads > 0 {
  truncated = false
}
```